Modules/_elementtree.c

```diff
@@ -908,13 +916,20 @@ deepcopy(elementtreestate *st, PyObject *object, PyObject *memo)
                     break;
                 }
             }
-            if (simple)
+            if (simple) {
+                // This does not call object.__copy__(), so it's still safe.
```

Comment:
This does not call any user code. It does not even call `__hash__` and `__eq__` for keys, because they are exact strings. Keys and values can be shared, not copied, because they are exact strings.